src/lib/operators/join_sort_merge/column_materializer.hpp

```diff
@@ -30,9 +30,17 @@ using MaterializedSegment = std::vector<MaterializedValue<T>>;
 template <typename T>
 using MaterializedSegmentList = std::vector<std::shared_ptr<MaterializedSegment<T>>>;
 
+template <typename T>
+struct SampleRequest {
+  explicit SampleRequest(uint32_t sample_count) : samples_to_collect(sample_count), collected_samples() {}
+
+  const uint32_t samples_to_collect;
```

Comment:
why uint32?

Reply:
That wasn't my point. We shouldn't use naked data types without documenting why they are the best fit, or better, having a name for them.

Reply:
A name would probably be an overkill here. I thought the name and the type are self-explanatory.  Any ideas how to improve that?

Reply:
Changed to `SubSample` and using ChunkOffset now.